app/Policies/UserPolicy.php

```diff
@@ -8,71 +8,115 @@
-     * authorizing application administrators to perform any action.
+     * Determine whether the user can view the list of users.
+     *
+     * @param  \App\User  $user
+     * @return mixed
      */
-    public function before($user, $ability)
+    public function index(User $user)
     {
-        // TODO: Admins...
-        // if ($user->isSuperAdmin()) {
-        //     return true;
-        // }
-        return false;
+        return $user->isAdmin();
     }
-
+    
     /**
-     * Determine whether the user can view the user.
+     * Determine whether the user can view the create users page.
      *
      * @param  \App\User  $user
-     * @param  \App\User  $user
      * @return boolean
      */
-    public function view(User $user, User $user2)
+    public function create(User $user)
     {
-        // TODO: Check if user role is > 0
-        return $user->role > 1 || $user->id === $user2->id;
+        return $user->isAdmin();
     }
-
+    
     /**
-     * Determine whether the user can create users.
+     * Determine whether the user can create a user.
      *
      * @param  \App\User  $user
      * @return boolean
      */
-    public function create(User $user)
+    public function store(User $user)
     {
-        // TODO: Check if user role > 1
-        return true;
+        return $user->isAdmin();
     }
 
     /**
-     * Determine whether the user can update the user2.
+     * Determine whether the user can view the user.
+     *
+     * @param  \App\User  $user the current user
+     * @param  \App\User  $user2 the user to be viewed
+     * @return boolean
+     */
+    public function show(User $user, User $user2)
+    {
+        if ($user->isAdmin())
+            return true;
+        else
+            return $user->isUser() && $user->id === $user2->id;
+    }
+    
+    /**
+     * Determine whether the user can view the edit page for a user.
      *
      * @param  \App\User  $user
+     * @param  \App\User  $user2 the user to be edited
+     * @return mixed
+     */
+    public function edit(User $user, User $user2)
+    {
+        if ($user->isAdmin())
+            return true;
+        else
+            return $user->isUser() && $user->id === $user2->id;
+    }
+
+    /**
+     * Determine whether the user can update user2.
+     *
      * @param  \App\User  $user
+     * @param  \App\User  $user2
      * @return boolean
      */
     public function update(User $user, User $user2)
     {
-        // Users can update themselves
-        //return $user->id === $user->id;
-        return true;
+        if ($user->isAdmin())
+            return true;
+        else
+            return $user->isUser() && $user->id === $user2->id;
+    }
+    
+    /**
+     * Determine whether the user can update a user's role.
+     *
+     * @param  \App\User  $user
+     * @return boolean
+     */
+    public function updateRole(User $user)
+    {
+        return $user->isAdmin();
```

Comment:
Managers need to be able to update user roles to allow them on the site. This check should make sure that the value they are setting is not greater than or equal to their own role.

Reply:
From talking to Dev it sounded like they only want admins to be able to do this. Also we might have issues if we are revealing student information to other students since managers would gain access to the users datatable.